kb(drawer):Click in dropdown closes the Drawer (#1101)
telerik/blazor-docs: knowledge-base/drawer-click-in-dropdown-closes-drawer.md

---
title: Click in dropdown closes the Drawer
description: Click in dropdown closes the Drawer
type: troubleshooting
page_title: Click in dropdown closes the Drawer
slug: drawer-kb-click-in-dropdown-closes-drawer
position: 
tags: drawer, dropdown, click, close
ticketid: 1575100
res_type: kb
---

## Environment

<table>
    <tr>
        <td>Product</td>
        <td>Drawer for Blazor</td>
    </tr>
    <tr>
        <td>Product</td>
        <td>AutoComplete for Blazor, ComboBox for Blazor, DropDownList for Blazor, MultiSelect for Blazor</td>
    </tr>
    <tr>
        <td>Product</td>
        <td>DatePicker for Blazor, DateRangePicker for Blazor, DateTimePicker for Blazor, DateTimePicker for Blazor</td>
    </tr>

> </table>

Reviewer: This structure with repeating Product looks a bit confusing. Can we present the information in a bit different way?

## Description

I have a DropDownList inside a Drawer. The Drawer uses the Overlay mode. When I click in the dropdown to select an item, the Drawer closes. Click and drag the scrollbar in the dropdown also closes the Drawer.

The same behavior occurs if I place a DatePicker inside a Drawer - click in the calendar popup closes the Drawer.

How I can prevent this? How to keep the Drawer open when selecting items in the dropdown and/or calendar?

## Cause\Possible Cause(s)

> By design, the popup elements of our components are not rendered in their place of declaration but on root level. Thus, when you are using a [DropDownList]({%slug components/dropdownlist/overview%}), for example, its popup is not technically part of the DropDownList component, it is rendered outside of it.

Reviewer: ```suggestion
By design, the popup elements of our components are not rendered in their place of declaration but at root level in the Razor component tree. Thus, when you are using a [DropDownList]({%slug components/dropdownlist/overview%}), for example, its popup is not technically part of the DropDownList component, it is rendered outside of it.
```

> As the popup of the DropDownList is rendered on root level, it is practically outside of the Drawer as well. So, when you click on an item in the dropdown, you are basically clicking outside of the Drawer. By design of its [Overlay mode]({%slug drawer-modes%}#overlay-mode), clicking outside of the Drawer container is expected to close the Drawer.

Reviewer: ```suggestion
As the popup of the DropDownList is rendered at root level, it is practically outside of the Drawer as well. So, when you click on an item in the dropdown, you are basically clicking outside of the Drawer. By design in [Overlay mode]({%slug drawer-modes%}#overlay-mode), clicking outside of the Drawer container is expected to close the Drawer.
```

You can verify this yourself using your dev tools to inspect the rendering. You can also test selecting an item from the DropDownList with keyboard navigation only (no click in the popup) - as a result, the Drawer will remain opened as you haven't clicked outside of it, in the DropDownList popup. The same behavior can be observed if you scroll without a click in the scrollbar.

## Solution

> To handle the scenario, do one of the following:

Reviewer: Provide the two options here and link via anchors. In this way, readers who immediately reject one option will see and navigate to the other one.

### Use the Push Drawer Mode 

You may try the Push mode of the Drawer. When this mode is enabled, clicking outside of the Drawer container will not close it. 

This mode, however, introduces different appearance and positioning of the Drawer compared to the Overlay mode. If you'd like to keep the styles of the Overlay mode but prevent the Drawer from closing on click in the popup, try the following:

* Select the [Push mode]({%slug drawer-modes%}#push-mode).
* Customize the Drawer to incorporate the styles of the Overlay mode.

> The exmple below demonstrates the described approach. In this case, the Drawer will not close when you click outside of it, so you may consider some UI option for that. In the sample below, the Drawer contains a close button, for instance.

Reviewer: ```suggestion
The example below demonstrates the described approach. In this case, the Drawer will not close when you click outside of it, so you may consider some UI option for that. In the sample below, the Drawer contains a close button, for instance.
```
